frontend/public/components/factory/modal.tsx

```diff
@@ -118,15 +118,27 @@ export const ModalSubmitFooter: React.SFC<ModalSubmitFooterProps> = ({
   cancelText,
   submitDisabled,
   submitDanger,
+  resetText,
+  reset,
 }) => {
   const onCancelClick = (e) => {
     e.stopPropagation();
     cancel(e);
   };
 
+  const onResetClick = (e) => {
+    e.stopPropagation();
+    reset(e);
+  };
+
   return (
     <ModalFooter inProgress={inProgress} errorMessage={errorMessage} message={message}>
       <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
+        {resetText && (
```

Comment:
It would be better to check for `reset` and then just have generic `Reset` as the text if `resetText` was not provided.

Reply:
done